src/components/accessories/billTable/BillTable.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { FC, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { Link } from "react-router-dom";
+import { isJsxAttributes } from "typescript";
+import { FullBillDTO } from "../../../generated";
+import { renderDate } from "../../../libraries/formatUtils/dataFormatting";
+import { getPendingBills, searchBills } from "../../../state/bills/actions";
+import { IState } from "../../../types";
+import { CustomModal } from "../customModal/CustomModal";
+import Table from "../table/Table";
+import RenderBillDetails from "./RenderBillDetails";
+type IStatus = "ALL" | "PENDING" | "CLOSE" | "DELETE";
+interface IBillTableProps {
+  status: IStatus;
+  fromDate: string;
+  toDate: string;
+  patientCode: number;
+}
+export const BillTable: FC<IBillTableProps> = ({
+  status,
+  fromDate,
+  toDate,
+  patientCode,
+}) => {
+  const { t } = useTranslation();
+  const header = ["date", "patient", "balance", "status"];
+  const label = {
+    id: t("bill.code"),
+    date: t("bill.date"),
+    patient: t("bill.patient"),
+    status: t("bill.status"),
+    amount: t("bill.amount"),
+    balance: t("bill.balance"),
+  };
+  const order = ["date", "balance"];
+  const dispatch = useDispatch();
+  const [fullBill, setFullBill] = useState({} as FullBillDTO);
+
+  useEffect(() => {
+    switch (status) {
+      case "PENDING":
+        dispatch(getPendingBills(patientCode));
+        break;
+      case "CLOSE":
+        dispatch(searchBills(fromDate, toDate, patientCode));
+        break;
+      case "DELETE":
+        dispatch(searchBills(fromDate, toDate, patientCode));
+        break;
+      case "ALL":
+        dispatch(searchBills(fromDate, toDate, patientCode));
+        break;
+    }
+  }, [status, fromDate, toDate, patientCode]);
+
+  const data = useSelector<IState, FullBillDTO[]>(
+    (state) => state.bills.searchBills.data ?? []
+  );
+
+  const formatDataToDisplay = (data: FullBillDTO[] | undefined) => {
+    let results = new Array();
+    if (data)
+      results = data.map((item) => {
+        return {
+          id: item.billDTO?.id ?? "",
+          date: item.billDTO?.date ? renderDate(item.billDTO.date) : "",
+          patient: (
+            <Link
+              to={`/details/${item.billDTO?.patientDTO?.code}/edit`}
+              style={{ textDecoration: "none" }}
+            >
+              <strong>{item.billDTO?.patName}</strong>
+            </Link>
+          ),
+          amount: item.billDTO?.amount,
+          balance: item.billDTO?.balance,
+          status: switchStatus(item.billDTO?.status),
+        };
+      });
+    if (status === "CLOSE") {
+      results.filter((item) => item.status === "C");
+    }
+    if (status === "DELETE") {
+      results.filter((item) => item.status === "D");
+    }
+    return results;
+  };
```

Comment:
I whish to put this under a useEffect statement in order to filter the results when the status variable changes.

So you won't filter / format data every time the table renderds but only when the status and the data will do.

Plus, the data format I will put inside a lib.ts in order to keep this useEffect as much clean as possibile.

Reply:
There is something wrong with the searchBills action: when I acess to ALL tab the table don't display the data. It displays only when I switch betweens the tabs. Maybe my previous suggestion could lead to solve this, in any case just keep the eyes on this issue.

Reply:
I resolved the problem. I created another reducer for pending bills.